### PR TITLE
refactor(cloud): migrate hosts and zones data source to plugin framework

### PR DESCRIPTION
**Pull request by villevsv-upcloud**

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -7,12 +7,12 @@ require (
 	github.com/hashicorp/go-cty v1.4.1-0.20200414143053-d3edf31b6320
 	github.com/hashicorp/go-retryablehttp v0.7.7
 	github.com/hashicorp/go-uuid v1.0.3
-	github.com/hashicorp/terraform-plugin-framework v1.7.0
+	github.com/hashicorp/terraform-plugin-framework v1.4.1
```
Comment (kangasta) [inline go.mod:10]: Are these downgrades included here on purpose or leftovers from #571?

File: internal/service/cloud/zones_data_source.go [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+	zones, err := d.client.GetZones(ctx)
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Unable to read zones",
+			utils.ErrorDiagnosticDetail(err),
+		)
+		return
+	}
+
+	data.ID = types.StringValue(time.Now().UTC().String())
```
Comment (kangasta) [inline internal/service/cloud/zones_data_source.go:89]: Would this work with `"zones" + data.FilterType` as the ID 🤔 There should not be need for multiple instances of zones list in the state. Not sure how that would work with modules, though.

Anyway, not in the scope of this PR, as we have similar solution for all generic data-sources at the moment.

**Review by kangasta — APPROVED** [review]
Could remove the dependency version downgrades, if those are not needed, but otherwise LGTM!

File: go.mod (hunk 1 above)
Comment (villevsv-upcloud) [reply go.mod]: Leftovers, removed now! :+1:

File: internal/service/cloud/zones_data_source.go (hunk 2 above)
Comment (villevsv-upcloud) [reply internal/service/cloud/zones_data_source.go:89]: That sounds like a good idea.

**Pull request merged**